### PR TITLE
Implement gossip topic scoring params

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

+ gossip v1.1 has an implementation for scoring params but we only use default params

**Description**

+ go with lighthouse's implementation at https://github.com/sigp/lighthouse/blob/b0ac3464ca5fb1e9d75060b56c83bfaf990a3d25/beacon_node/eth2_libp2p/src/behaviour/gossipsub_scoring_parameters.rs#L83

part of #3084

### CODE REVIEW

File: packages/lodestar/src/network/gossip/scoringParameters.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+  (MAX_IN_MESH_SCORE + MAX_FIRST_MESSAGE_DELIVERIES_SCORE) *
+  (BEACON_BLOCK_WEIGHT +
+    +BEACON_AGGREGATE_PROOF_WEIGHT +
+    beaconAttestationSubnetWeight * ATTESTATION_SUBNET_COUNT +
+    VOLUNTARY_EXIT_WEIGHT +
+    PROPOSER_SLASHING_WEIGHT +
+    ATTESTER_SLASHING_WEIGHT);
+
+/**
+ * The following params is implemented by Lighthouse.
```
Comment (dapplion) [inline packages/lodestar/src/network/gossip/scoringParameters.ts]: Can you link to exact code where this is taken from?

File: packages/lodestar/test/unit/network/gossip/scoringParameters.ts [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+import {expect} from "chai";
+import {stringifyGossipTopic} from "../../../../src/network/gossip/topic";
+import {GossipType} from "../../../../src/network";
+import {TopicScoreParams} from "libp2p-gossipsub/src/score";
+
+/**
+ * Refer to Teku tests at
+ * https://github.com/ConsenSys/teku/blob/e18ab9903442410aa04b590c4cc46734e13d3ffd/networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/config/GossipScoringConfiguratorTest.java#L38
+ */
+export function runGossipScoringParametersTests(): void {
```
Comment (twoeths) [inline packages/lodestar/test/unit/network/gossip/scoringParameters.ts]: this test needs to run with mainnet profile so I have to call it through `mainnetTest.ts`, any better ways to do it?

Comment (dapplion) [reply packages/lodestar/test/unit/network/gossip/scoringParameters.ts]: Yeah, can you place the tests under `test/unit-mainnet` for consistency with other tests? Then add to package json
```
"test:unit": "LODESTAR_PRESET=mainnet mocha 'test/unit-mainnet/**/*.test.ts'"
```

The minimal preset is injected in this script for reference https://github.com/ChainSafe/lodestar/blob/7beccf0128c94d9e1720d4e844a53035ff4975ea/packages/lodestar/.mocharc.yaml#L5

**Comment by twoeths** [comment]
issue #3132 found during testing this branch, however it does not look like that comes from this branch itself. I can't reproduce it in my local environment or contabo server.

File: packages/lodestar/src/chain/chain.ts [hunk 3]
```diff
@@ -38,6 +38,7 @@ import {ForkDigestContext, IForkDigestContext} from "../util/forkDigestContext";
 import {LightClientIniter} from "./lightClient";
 import {AggregatedAttestationPool} from "./opPools/aggregatedAttestationPool";
 import {Archiver} from "./archiver";
+import {Eth2Context} from ".";
```
Comment (dapplion) [inline packages/lodestar/src/chain/chain.ts]: Import explicitly from module

File: packages/lodestar/src/network/gossip/scoringParameters.ts [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+};
+
+type MeshMessageInfo = {
+  decaySlots: number;
+  capFactor: number;
+  activationWindow: number;
+  currentSlot: number;
+};
+
+export class GossipPeerScoreParamsBuilder {
```
Comment (dapplion) [inline packages/lodestar/src/network/gossip/scoringParameters.ts]: This should not be a class, it holds not state and is only called one. Please convert to function that returns the params

File: packages/lodestar/src/chain/chain.ts [hunk 5]
```diff
@@ -136,6 +138,11 @@ export class BeaconChain implements IBeaconChain {
       opts,
     });
 
+    this.eth2Context = {
```
Comment (dapplion) [inline packages/lodestar/src/chain/chain.ts]: Don't attach this data to the state, it's unnecessary. Just pass the Eth2Context down to gossipsub

**Review by dapplion — APPROVED** [review]
Looks good! Overall structure and approach is sound but I have not verified all numbers and constants.

**Review by wemeetagain — APPROVED** [review]
Lgtm

**Pull request merged**